src/elements/earn/portfolio/poolTokens/poolTokensTable/PoolTokensTable.tsx

```diff
@@ -1,42 +1,70 @@
 import { ReactComponent as IconSearch } from 'assets/icons/search.svg';
 import { DataTable, TableColumn } from 'components/table/DataTable';
 import { useMemo, useState } from 'react';
+import { useAppSelector } from 'redux/index';
+import { PoolToken } from 'services/observables/tokens';
+import { prettifyNumber } from 'utils/helperFunctions';
+import { PoolTokensCellActions } from './PoolTokensCellActions';
+import { PoolTokensCellName } from './PoolTokensCellName';
+import { PoolTokensCellReserve } from './PoolTokensCellReserve';
 
 export const PoolTokensTable = () => {
   const [search, setSearch] = useState('');
+  const poolTokens = useAppSelector<PoolToken[]>(
```

Comment:
suggestion: extract lines 13-71 to a custom hook `usePoolTokenColumns` that returns data & columns